src/figuresClasses.ts

```diff
@@ -1,19 +1,96 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 export interface Figure {
-
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape;
+
+  color: Color;
+
+  a: number;
+
+  b: number;
+
+  c: number;
 
+  constructor(color: Color, a: number, b: number, c: number) {
+    const max = Math.max(a, b, c);
+
+    if ((max >= (a + b + c - max)) || (a <= 0) || (b <= 0) || (c <= 0)) {
+      throw new Error(`sides ${a}, ${b} and ${c} can't form a triangle`);
+    }
+
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const s: number = (this.a + this.b + this.c) / 2;
```

Comment:
You don't need to add types for calculations. It'll be a number by default. You can remove it everywhere 
```suggestion
    const s = (this.a + this.b + this.c) / 2;
```